### PR TITLE
For TestCafe i2195 issue

### PR DESCRIPTION
**Pull request by miherlosev**

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/client/utils/check-by-condition.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import Promise from 'pinkie';
+import nativeMethods from '../sandbox/native-methods';
+/*eslint-disable no-unused-vars*/
+import { ICheckByConditionOptions } from '../../typings/client';
+/*eslint-enable no-unused-vars*/
+
+const DEFAULT_CHECK_CONDITION_TIMEOUT: number = 100;
+const DEFAULT_ABORT_TIMEOUT: number           = 5000;
+
+export default function (predicate: Function, options: ICheckByConditionOptions | null): Promise<any> {
```
Comment (LavrovArtem) [inline src/client/utils/check-by-condition.ts]: `export default function (predicate: Function, options: ICheckByConditionOptions = {}): Promise<void>`

File: src/client/utils/check-by-condition.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    options                       = options || {};
```
Comment (LavrovArtem) [inline src/client/utils/check-by-condition.ts]: remove

File: src/client/utils/check-by-condition.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        let checkIntervalId: number = null;
+        let abortTimeoutId: number  = null;
+
+        const clearTimeouts = () => {
+            nativeMethods.clearInterval.call(options.win, checkIntervalId);
+            nativeMethods.clearTimeout.call(options.win, abortTimeoutId);
+        };
+
+        abortTimeoutId = nativeMethods.setTimeout.call(options.win, () => {
+            clearTimeouts();
```
Comment (LavrovArtem) [inline src/client/utils/check-by-condition.ts]: `nativeMethods.clearInterval.call(options.win, checkIntervalId);`

File: src/client/utils/check-by-condition.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            reject();
+        }, options.abortAfterMs);
+
+        checkIntervalId = nativeMethods.setInterval.call(options.win, () => {
+            const result = predicate();
+
+            if (result)
```
Comment (LavrovArtem) [inline src/client/utils/check-by-condition.ts]:
```js
            nativeMethods.clearInterval.call(options.win, checkIntervalId);
            nativeMethods.clearTimeout.call(options.win, abortTimeoutId);
```

File: src/client/utils/check-by-condition.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+/*eslint-disable no-unused-vars*/
+import { ICheckByConditionOptions } from '../../typings/client';
+/*eslint-enable no-unused-vars*/
+
+const DEFAULT_CHECK_CONDITION_TIMEOUT: number = 100;
+const DEFAULT_ABORT_TIMEOUT: number           = 5000;
+
+export default function (predicate: Function, options: ICheckByConditionOptions | null): Promise<any> {
+    options                       = options || {};
+    options.checkConditionEveryMs = options.checkConditionEveryMs || DEFAULT_CHECK_CONDITION_TIMEOUT;
```
Comment (LavrovArtem) [inline src/client/utils/check-by-condition.ts]:
```js
const checkConditionEveryMs = options.checkConditionEveryMs || DEFAULT_CHECK_CONDITION_TIMEOUT;
const abortAfterMs          = options.abortAfterMs || DEFAULT_ABORT_TIMEOUT;
const win                   = options.win || window;
```

File: src/client/utils/check-by-condition.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    options.abortAfterMs          = options.abortAfterMs || DEFAULT_ABORT_TIMEOUT;
+    options.win                   = options.win || window;
+
+    return new Promise<any>((resolve: Function, reject: Function) => {
```
Comment (LavrovArtem) [inline src/client/utils/check-by-condition.ts]: `Promise<void>`

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
These changes is not compatible with the current Testcafe.

Review please.

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
This [change](https://github.com/DevExpress/testcafe-hammerhead/pull/1974/files#diff-1b88cb0c1c49b257c71d2ece77bc3e83R85) doesnot compatible with the current TestCafe. 
Review please.

**Pull request merged**